javafx doesn't work on mac os x

This was run on f0a5833e6b872e1511ea3d1bd43c69680e37b047


<img width="589" alt="Screen Shot 2021-05-05 at 9 03 35 AM" src="https://user-images.githubusercontent.com/3514957/117153737-dd96d180-ad80-11eb-9c96-876fc8fc4cc3.png">

The likely culprit seems to be the new chip architecture that is used in M1 (arm64)


```
[error] May 05, 2021 9:02:36 AM com.sun.javafx.application.PlatformImpl startup
[error] WARNING: Unsupported JavaFX configuration: classes were loaded from 'unnamed module @5fc3f96c'
[info] 2021-05-05T14:02:36UTC INFO [NeutrinoNode] Starting sync node, height=44000 hash=000000000122898b31073a770a97cf599c00672fc8d6ae15652235862f8b76d8
[info] 2021-05-05T14:02:36UTC INFO [BitcoinSServerMain] Done starting Main!
[error] Loading library prism_es2 from resource failed: java.lang.UnsatisfiedLinkError: /Users/chris/.openjfx/cache/17-ea/libprism_es2.dylib: dlopen(/Users/chris/.openjfx/cache/17-ea/libprism_es2.dylib, 1): no suitable image found.  Did find:
[error] 	/Users/chris/.openjfx/cache/17-ea/libprism_es2.dylib: mach-o, but wrong architecture
[error] 	/Users/chris/.openjfx/cache/17-ea/libprism_es2.dylib: mach-o, but wrong architecture
[error] java.lang.UnsatisfiedLinkError: /Users/chris/.openjfx/cache/17-ea/libprism_es2.dylib: dlopen(/Users/chris/.openjfx/cache/17-ea/libprism_es2.dylib, 1): no suitable image found.  Did find:
[error] 	/Users/chris/.openjfx/cache/17-ea/libprism_es2.dylib: mach-o, but wrong architecture
[error] 	/Users/chris/.openjfx/cache/17-ea/libprism_es2.dylib: mach-o, but wrong architecture
[error] 	at java.base/java.lang.ClassLoader$NativeLibrary.load0(Native Method)
[error] 	at java.base/java.lang.ClassLoader$NativeLibrary.load(ClassLoader.java:2442)
[error] 	at java.base/java.lang.ClassLoader$NativeLibrary.loadLibrary(ClassLoader.java:2498)
[error] 	at java.base/java.lang.ClassLoader.loadLibrary0(ClassLoader.java:2694)
[error] 	at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2627)
[error] 	at java.base/java.lang.Runtime.load0(Runtime.java:768)
[error] 	at java.base/java.lang.System.load(System.java:1837)
[error] 	at com.sun.glass.utils.NativeLibLoader.installLibraryFromResource(NativeLibLoader.java:214)
[error] 	at com.sun.glass.utils.NativeLibLoader.loadLibraryFromResource(NativeLibLoader.java:194)
[error] 	at com.sun.glass.utils.NativeLibLoader.loadLibraryInternal(NativeLibLoader.java:135)
[error] 	at com.sun.glass.utils.NativeLibLoader.loadLibrary(NativeLibLoader.java:53)
[error] 	at com.sun.prism.es2.ES2Pipeline.lambda$static$0(ES2Pipeline.java:62)
[error] 	at java.base/java.security.AccessController.doPrivileged(Native Method)
[error] 	at com.sun.prism.es2.ES2Pipeline.<clinit>(ES2Pipeline.java:51)
[error] 	at java.base/java.lang.Class.forName0(Native Method)
[error] 	at java.base/java.lang.Class.forName(Class.java:315)
[error] 	at com.sun.prism.GraphicsPipeline.createPipeline(GraphicsPipeline.java:218)
[error] 	at com.sun.javafx.tk.quantum.QuantumRenderer$PipelineRunnable.init(QuantumRenderer.java:91)
[error] 	at com.sun.javafx.tk.quantum.QuantumRenderer$PipelineRunnable.run(QuantumRenderer.java:124)
[error] 	at java.base/java.lang.Thread.run(Thread.java:829)
[error] Loading library prism_sw from resource failed: java.lang.UnsatisfiedLinkError: /Users/chris/.openjfx/cache/17-ea/libprism_sw.dylib: dlopen(/Users/chris/.openjfx/cache/17-ea/libprism_sw.dylib, 1): no suitable image found.  Did find:
[error] 	/Users/chris/.openjfx/cache/17-ea/libprism_sw.dylib: mach-o, but wrong architecture
[error] 	/Users/chris/.openjfx/cache/17-ea/libprism_sw.dylib: mach-o, but wrong architecture
[error] java.lang.UnsatisfiedLinkError: /Users/chris/.openjfx/cache/17-ea/libprism_sw.dylib: dlopen(/Users/chris/.openjfx/cache/17-ea/libprism_sw.dylib, 1): no suitable image found.  Did find:
[error] 	/Users/chris/.openjfx/cache/17-ea/libprism_sw.dylib: mach-o, but wrong architecture
[error] 	/Users/chris/.openjfx/cache/17-ea/libprism_sw.dylib: mach-o, but wrong architecture
[error] 	at java.base/java.lang.ClassLoader$NativeLibrary.load0(Native Method)
[error] 	at java.base/java.lang.ClassLoader$NativeLibrary.load(ClassLoader.java:2442)
[error] 	at java.base/java.lang.ClassLoader$NativeLibrary.loadLibrary(ClassLoader.java:2498)
[error] 	at java.base/java.lang.ClassLoader.loadLibrary0(ClassLoader.java:2694)
[error] 	at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2627)
[error] 	at java.base/java.lang.Runtime.load0(Runtime.java:768)
[error] 	at java.base/java.lang.System.load(System.java:1837)
[error] 	at com.sun.glass.utils.NativeLibLoader.installLibraryFromResource(NativeLibLoader.java:214)
[error] 	at com.sun.glass.utils.NativeLibLoader.loadLibraryFromResource(NativeLibLoader.java:194)
[error] 	at com.sun.glass.utils.NativeLibLoader.loadLibraryInternal(NativeLibLoader.java:135)
[error] 	at com.sun.glass.utils.NativeLibLoader.loadLibrary(NativeLibLoader.java:53)
[error] 	at com.sun.prism.sw.SWPipeline.lambda$static$0(SWPipeline.java:42)
[error] 	at java.base/java.security.AccessController.doPrivileged(Native Method)
[error] 	at com.sun.prism.sw.SWPipeline.<clinit>(SWPipeline.java:41)
[error] 	at java.base/java.lang.Class.forName0(Native Method)
[error] 	at java.base/java.lang.Class.forName(Class.java:315)
[error] 	at com.sun.prism.GraphicsPipeline.createPipeline(GraphicsPipeline.java:218)
[error] 	at com.sun.javafx.tk.quantum.QuantumRenderer$PipelineRunnable.init(QuantumRenderer.java:91)
[error] 	at com.sun.javafx.tk.quantum.QuantumRenderer$PipelineRunnable.run(QuantumRenderer.java:124)
[error] 	at java.base/java.lang.Thread.run(Thread.java:829)
[error] Graphics Device initialization failed for :  es2, sw
[error] Error initializing QuantumRenderer: no suitable pipeline found
[error] java.lang.RuntimeException: java.lang.RuntimeException: Error initializing QuantumRenderer: no suitable pipeline found
[error] 	at com.sun.javafx.tk.quantum.QuantumRenderer.getInstance(QuantumRenderer.java:280)
[error] 	at com.sun.javafx.tk.quantum.QuantumToolkit.init(QuantumToolkit.java:245)
[error] 	at com.sun.javafx.tk.Toolkit.getToolkit(Toolkit.java:262)
[error] 	at com.sun.javafx.application.PlatformImpl.startup(PlatformImpl.java:286)
[error] 	at com.sun.javafx.application.PlatformImpl.startup(PlatformImpl.java:160)
[error] 	at com.sun.javafx.application.LauncherImpl.startToolkit(LauncherImpl.java:661)
[error] 	at com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:681)
[error] 	at com.sun.javafx.application.LauncherImpl.lambda$launchApplication$2(LauncherImpl.java:198)
[error] 	at java.base/java.lang.Thread.run(Thread.java:829)
[error] Caused by: java.lang.RuntimeException: Error initializing QuantumRenderer: no suitable pipeline found
[error] 	at com.sun.javafx.tk.quantum.QuantumRenderer$PipelineRunnable.init(QuantumRenderer.java:94)
[error] 	at com.sun.javafx.tk.quantum.QuantumRenderer$PipelineRunnable.run(QuantumRenderer.java:124)
[error] 	... 1 more

```

--- a/project/Deps.scala
+++ b/project/Deps.scala
@@ -100,13 +100,20 @@ object Deps {
     val scalaFx =
       "org.scalafx" %% "scalafx" % V.scalaFxV withSources () withJavadoc ()
 
+    lazy val arch = System.getProperty("os.arch")
     lazy val osName = System.getProperty("os.name") match {
       case n if n.startsWith("Linux")   => "linux"
-      case n if n.startsWith("Mac")     => "mac"
+      case n if n.startsWith("Mac")     =>
+        if (arch == "aarch64" ) {
+          //needed to accommodate the different chip
+          //arch for M1
+          s"mac-${arch}"
+        } else {
+          "mac"
+        }
       case n if n.startsWith("Windows") => "win"
-      case _                            => throw new Exception("Unknown platform!")
+      case x                            => throw new Exception(s"Unknown platform $x!")
     }
-
     // Not sure if all of these are needed, some might be possible to remove
     lazy val javaFxBase =
       "org.openjfx" % s"javafx-base" % V.javaFxV classifier osName withSources () withJavadoc ()

